site/learn/expert-advice/identity-basics/registration-best-practices.md

```diff
@@ -0,0 +1,140 @@
+
+## Ease the pain of registration
+
+If you've decided you need a registration form, remember that it is a form first and foremost. You should follow known best practices.
+
+### A form is a form is a form
+
+The Nielsen Norman Group discusses improving forms in their 2016 article, [Website Forms Usability: Top 10 Recommendations](https://www.nngroup.com/articles/web-form-design/). The number one suggestion is:
+
+> "Keep it short. ... Eliminating unnecessary fields requires more time [to decide what data is worth asking for], but the reduced user effort and increased completion rates make it worthwhile. Remove fields which collect information that can be (a) derived in some other way, (b) collected more conveniently at a later date, or (c) simply omitted."
+
+Carefully consider the information you are asking for. The fewer fields the better. While admittedly in a different context, [Imagescape more than doubled a contact form conversion rate](https://unbounce.com/conversion-rate-optimization/how-to-optimize-contact-forms/) by decreasing the number of fields from 11 to 4. There may be data needed only for certain features of your application; ask for it when that feature is first accessed, rather than at signup. 
+
+Make sure your form is mobile friendly; test at various screen sizes. The tediousness of data entry on a mobile device and the prevalence of their usage are another reason to have as few signup form fields as possible.
+
+If a form field is optional, clearly mark it so. Even better, don't ask for optional data on the initial user registration. Request that information later, when the user is more engaged and has discovered the value of your offering.
+
+Provide clear error messages when data fails to validate. Use both client side validation, which is faster, and server side validation, which is tamper proof. On the topic of tampering, ensure any form is submitted over TLS. You want to keep submitted information confidential and secure.
+
+Make use of the full suite of HTML elements. Dropdowns and radio buttons are powerful, but number and email input fields leverage browsers' built-in validation and should be used as well. If you aren't sure what's supported, use tools like [caniuse.com](https://caniuse.com/) to verify compatibility.
+
+### Registration forms are unique
+
+But registration forms aren't just another form. They are the gateway to your full application or site.
+
+What causes angst when a user is signing up? This [2012 paper (PDF)](https://discovery.ucl.ac.uk/id/eprint/1378346/1/ewic_hci12_diss_paper7.pdf) examined registration for government services. It defined sign up friction as "as the imbalance between the business process (user goals) and [required] security behaviour" around signing up. This study found friction was best explained by the following attributes of a signup process:
```

Comment:
It defined sign up friction as "~as~ the imbalance between..."

Reply:
Awesome, thanks!